### ISSUE TITLE
useEventListenerOutside and focus loss

### ISSUE THREAD

**Issue by salper** [reakit/reakit#677]
### Current behavior

When a nested dialog is `modal: false`, and there's a focus loss, the focus is given to an element outside of the nested modal, calling hide [here](https://github.com/reakit/reakit/blob/239eb5622a1a02cd6f69c857bb725c8250dad155/packages/reakit/src/Dialog/__utils/useHideOnClickOutside.ts#L61). It's probably due to the fact that for the browser, the parent dialog focusable elements are on the same "plane". For context, the bug occured on my application when the activeElement was unmounted.

### Steps to reproduce the bug

https://codesandbox.io/s/nice-cloud-xo9th using chrome.

### Expected behavior

The nested dialog should not close (ie if reakit still relies on focus event for this, the nested dialog container should get the focus on focus loss).

### Possible solutions

As it's a focus loss problem, I've tried to play with tabIndex in order to force the browser `activeElement` selection, without success. Programmatically, clicking on the button, I can focus the nested dialog container using a ref, but it does not really scale. There's probably a reason for reakit to rely on the focus event (probably no focus trap), but that's the source of the issue.

### Environment

Please, run the command below inside your project directory.
```sh
$ npx envinfo --system --binaries --browsers --npmPackages "{react*,reakit*}"

  System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-3427U CPU @ 1.80GHz
    Memory: 825.36 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.17.0 - ~/.nvm/versions/node/v12.17.0/bin/node
    Yarn: 1.22.4 - ~/Code/eeselapp/eesel/node_modules/.bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.17.0/bin/npm
  Browsers:
    Brave Browser: 80.1.5.115
    Chrome: 83.0.4103.97
    Safari: 13.1.1
```

**Comment by Guria** [comment]
@salper tried to debug your example a bit.
Should I set a breakpoint on a [line that checks whether container contains target](https://github.com/reakit/reakit/blob/b3a20149338149257cebf63964d9b23eb3dbae57/packages/reakit/src/Dialog/__utils/useEventListenerOutside.ts#L60) it stops reproducing. Looks like the `target` of `focus` event is changing if there were a stop in a debugger in a `click` event handler.

So I changed your example to skip a tick, before calling blur:
`onClick={() => setImmediate(() => ref.current.blur())}`
And seems it also have the same effect. Not sure if it is what you are looking for, but hope it helps.

**Comment by salper** [comment]
@Guria Yeah, we can solve that programmatically, either skipping a tick or setting the focus. First time I've encountered the problem, I used `setTimeouts`, but it didn't always work (depending on the DOM context). If possible, the library should handle this case, but I have no idea how it could do.

**Comment by diegohaz** [comment]
Should be fixed in `v1.1.2`
